File: problems/easy/angry_professor/__tests__/test_angry_professor.js
const { angryProfessor } = require("../angry_professor");

const k = 3;
const a = [-1, -3, 4, 2];
const result = angryProfessor(k, a);

if (result !== "YES") {
    throw new Error(`❌ Test failed. Expected YES, got ${result}`);
}

console.log("✅ JavaScript test passed!");

File: problems/easy/angry_professor/__tests__/TestAngryProfessor.java
package problems.easy.angry_professor.__tests__;

import problems.easy.angry_professor.AngryProfessor;
import java.util.*;

public class TestAngryProfessor {
    public static void main(String[] args) {
        int k = 3;
        List<Integer> a = Arrays.asList(-1, -3, 4, 2);
        String result = AngryProfessor.angryProfessor(k, a);

        if (!result.equals("YES")) {
            throw new AssertionError("❌ Test failed. Expected YES, got " + result);
        }

        System.out.println("✅ Java test passed!");
    }
}
